src/gc-stacks.c

```diff
@@ -50,6 +50,16 @@ static void free_stack(void *stkbuf, size_t bufsz)
 
 #else
 
+# ifdef _OS_OPENBSD_
+static void *malloc_stack(size_t bufsz) JL_NOTSAFEPOINT
+{
+    void* stk = mmap(0, bufsz, PROT_READ | PROT_WRITE, MAP_PRIVATE | MAP_ANONYMOUS | MAP_STACK, -1, 0);
```

Comment:
Does OpenBSD automatically provide a stack guard page with MAP_STACK? If not seems like that handling is still required.

Reply:
stack guard is automatic on OpenBSD. `MAP_STACK` is used to let's the system know this memory chunck could be used as a stack. if it isn't present, the OS will refuse to use it as stack.